library/std/src/sys/locks/mutex/apple.rs

```diff
@@ -0,0 +1,47 @@
+//! Mutex for Apple platforms.
+//!
+//! On Apple platforms, priority inheritance is the default for locks. To avoid
+//! having to use pthread's mutex, which needs some tricks to work correctly, we
+//! instead use `os_unfair_lock`, which is small, movable and supports priority-
+//! inheritance and appeared with macOS 10.12, which is exactly our minimum
+//! supported version.
+
+use crate::cell::SyncUnsafeCell;
+
+// FIXME: move these definitions to libc
+
+#[allow(non_camel_case_types)]
+#[repr(C)]
+struct os_unfair_lock {
+    _opaque: u32,
+}
+
+const OS_UNFAIR_LOCK_INIT: os_unfair_lock = os_unfair_lock { _opaque: 0 };
+
+extern "C" {
+    fn os_unfair_lock_lock(lock: *mut os_unfair_lock);
```

Comment:
A blog from firefox states that os_unfair_lock has poor performance: https://hacks.mozilla.org/2022/10/improving-firefox-responsiveness-on-macos/

Is it still true?

(I do not deny that os_unfair_lock is better than current pthread implementation.)